src/providers/twitch/TwitchChannel.cpp

```diff
@@ -917,31 +898,16 @@ void TwitchChannel::refreshChatters()
         }
     }
 
-    // get viewer list
-    NetworkRequest("https://tmi.twitch.tv/group/user/" + this->getName() +
-                   "/chatters")
-
-        .onSuccess(
-            [this, weak = weakOf<Channel>(this)](auto result) -> Outcome {
-                // channel still exists?
-                auto shared = weak.lock();
-                if (!shared)
-                {
-                    return Failure;
-                }
-
-                auto data = result.parseJson();
-                this->chatterCount_ = data.value("chatter_count").toInt();
-
-                auto pair = parseChatters(std::move(data));
-                if (pair.first)
-                {
-                    this->updateOnlineChatters(pair.second);
-                }
-
-                return pair.first;
-            })
-        .execute();
+    // Get chatter list via helix api
+    getHelix()->getChatters(
+        this->roomId(),
+        getApp()->accounts->twitch.getCurrent()->getUserId(),
+        [this](HelixUserList *chatterList) {
```

Comment:
We need to use the same `weakOf` 'hack' that the previous TMI version used to ensure the callback doesn't attempt to use a TwitchChannel that has been deleted

Reply:
Not sure what you mean here. I tried to implement it based on the other functions and I think I got it...